docs(guide): add embeds guide (#124)
Cogmasters/concord: docs/guides/embeds.md

# Embeds

> Embeds in Discord are just **embedded content** that you can use to **organize the response of the bot**, with **title**, **image**, **footer**, and embed **color**. And as a benefit, it has a higher message content limit on its description.

Reviewer: Embeds in Discord are just **embedded content messages** that you can use to **organize the response of the bot**, with **larger titles**, **images**, **footers**, and embed **colors**. As an added benefit, they have a higher message content limit on its description. Embeds can be used to construct clickable dialogs, for example

## Concord & embeds

> **There are multiple ways of sending an embed with Concord**, but in this guide, we are using the `static approach` way.

Reviewer: **There are multiple ways of sending an embed with Concord**, but in this guide, we are using the "static approach" way.

# Embeds

> Embeds in Discord are just **embedded content messages** that you can use to **organize the response of the bot**, with **larger titles**, **images**, **footers**, and embed **colors**. As an added benefit, they have a higher message content limit on its description. Embeds can be used to construct clickable dialogs, for example.

Reviewer 2: ```suggestion
Discord's embeds stand for **embedded content messages** that can be used to present data with special formatting and structure. They have a higher message content character length limit as an added benefit and can be used to create clickable dialogs, for example.
```

Author: Just a question, can I add ** in `used to present data with special formatting and structure` and `higher message content character length limit`?

## Concord & embeds

> **There are multiple ways of sending an embed with Concord**, but in this guide, we are using the "static approach" way.

Reviewer 2: ```suggestion
**There are multiple ways of sending an embed with Concord**, but in this guide, we'll be focusing on the "static approach".
```

See the example below explaining all the fields of the structures:

```c
struct discord_embed embeds[] = {
    {
        .title = "Concord",
        .description = "Discord API library",
        .url = "https://github.com/Cogmasters/concord",
        .color = 0x3498DB,
        .timestamp = discord_timestamp(client),
        .footer =
            &(struct discord_embed_footer){
                .text = "github.com/Cogmasters/concord",
                .icon_url = "https://raw.githubusercontent.com/Cogmasters/concord/master/docs/static/concord-small.png",
            },
        .image =
            &(struct discord_embed_image){
                .url = "https://raw.githubusercontent.com/Cogmasters/concord/master/docs/static/social-preview.png",
            }
    },
};

struct discord_create_message params = {
    .content = "Cogmaster - Concord",
    .embeds =
        &(struct discord_embeds){
            .size = sizeof(embeds) / sizeof *embeds,
            .array = embeds,
        },
};
discord_create_message(client, msg->channel_id, &params, NULL);
```

> As you can see, first we are creating a structure with an array of the information of the embed we want the bot to send.
> 
> This is a simple example with only the most used fields.

Reviewer 2: ```suggestion
As you can see we are creating a structure with an array of embeds, which contain the information we want the bot to send.

*Keep in mind that this is a simple example, click [here](https://discord.com/developers/docs/resources/channel#embed-object) for a more thorough explanation of each embed field*
```

![Embed example](screenshots/embeds/embeds_example.png "Embed example")

[Embed structure](https://discord.com/developers/docs/resources/channel#embed-object):

* title: The title of the embed, the top part of the embed.
* description: The description/content of the embed.
* url: A URL that if someone clicks on the title of the embed, will redirect to it (NOTE: Title is required).
* color: A color, that can be represented by a binary number, HEX number, etc.
* timestamp: A `uint64_t` value with the current time, this value can be retrieved with [`discord_timestamp`](https://cogmasters.github.io/concord/group__DiscordClient.html#ga15a8fe1a6d3f30c18c6985b3afae11f5).

[Embed footer structure](https://discord.com/developers/docs/resources/channel#embed-object-embed-footer-structure):

> * text: A string with up to 2048 characters.

Reviewer 2: ```suggestion
* text: A string up to 2048 characters in length.
```

* icon_url: A valid URL that redirects to an image.

[Embed image structure](https://discord.com/developers/docs/resources/channel#embed-object-embed-image-structure):

* url: Same as footer->icon_URL.

> After creating the embed structure, now we will make the last structure so we can pass all the information needed to send a message.
> 
> The [`discord_create_message`](https://cogmasters.github.io/concord/structdiscord__create__message.html) uses the `struct discord_create_message params` to get the information of the message/embed and with it successfully (or maybe not) the message/embed to the channel.

Reviewer 2: ```suggestion
After creating the embed structure, now we will make the last structure so we can pass all the information needed to send a message.

The [`discord_create_message()`](https://cogmasters.github.io/concord/structdiscord__create__message.html) request expects its `params` to include the `embed` structure that will be sent over to the channel.
```

## Example bot

> This bot is just an example bot, with only the `static approach` way of sending an embed with Concord. For other and more detailed examples, you should take a look at this [example](examples/embed.c)

Reviewer 2: ```suggestion
The following example outlines the recommended `static approach` way for sending an embed. For alternatives, you may take a look at this [example](examples/embed.c)
```
